decoder.go

```diff
@@ -4,14 +4,54 @@ import (
 	"fmt"
 	"image"
 	"unsafe"
+
+	"github.com/pkg/errors"
 )
 
-// #cgo pkg-config: libavcodec libavutil libswscale
-// #include <libavcodec/avcodec.h>
-// #include <libavutil/imgutils.h>
-// #include <libswscale/swscale.h>
+/*
+#cgo pkg-config: libavcodec libavutil libswscale libavformat
+#include <libavcodec/avcodec.h>
+#include <libavutil/imgutils.h>
+#include <libswscale/swscale.h>
+#include <libavformat/avformat.h>
+#include <libavutil/avutil.h>
+
+int get_video_codec(AVFormatContext *avFormatCtx) {
```

Comment:
Thanks for adding the helper function. Much easier to follow than the previous version.

Let's check for nil pointer & return `AV_CODEC_ID_NONE` if it is called with nil pointer.

Reply:
If avformat_open_input and avformat_find_stream_info have succeeded, these pointers should be valid and non-NULL. 

Better safe than sorry! Added NULL checks.